File: app/src/main/java/de/nicidienase/chaosflix/entities/streaming/LiveConference.java
package de.nicidienase.chaosflix.entities.streaming;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by felix on 23.03.17.
 */

public class LiveConference {
	String conference;
	String slug;
	String author;
	String description;
	String keywords;
	String startsAt;
	String endsAt;
	List<Group> groups;

	public LiveConference() {}

	public LiveConference(String conference, String description) {
		this.conference = conference;
		this.description = description;
	}

	public String getConference() {
		return conference;
	}

	public void setConference(String conference) {
		this.conference = conference;
	}

	public String getSlug() {
		return slug;
	}

	public void setSlug(String slug) {
		this.slug = slug;
	}

	public String getAuthor() {
		return author;
	}

	public void setAuthor(String author) {
		this.author = author;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getKeywords() {
		return keywords;
	}

	public void setKeywords(String keywords) {
		this.keywords = keywords;
	}

	public String getStartsAt() {
		return startsAt;
	}

	public void setStartsAt(String startsAt) {
		this.startsAt = startsAt;
	}

	public String getEndsAt() {
		return endsAt;
	}

	public void setEndsAt(String endsAt) {
		this.endsAt = endsAt;
	}

	public List<Group> getGroups() {
		return groups;
	}

	public void setGroups(List<Group> groups) {
		this.groups = groups;
	}

	public static LiveConference getDummyObject(){
		LiveConference dummyCon = new LiveConference("DummyCon", "Conference McConferenceface");
		dummyCon.setGroups(new ArrayList<>());
		dummyCon.getGroups().add(Group.getDummyObject());
		dummyCon.setSlug("duco");
		dummyCon.setAuthor("");
		dummyCon.setDescription("A placeholder conference");
		dummyCon.setKeywords("");
		return dummyCon;
	}
}

File: common/src/main/java/de/nicidienase/chaosflix/common/entities/streaming/LiveConference.kt
package de.nicidienase.chaosflix.common.entities.streaming

import java.util.ArrayList

/**
 * Created by felix on 23.03.17.
 */

class LiveConference(
    var conference: String,
    var slug: String,
    var author: String,
    var description: String,
    var keywords: String,
    var startsAt: String,
    var endsAt: String,
    var groups: MutableList<Group>){

    constructor(conference: String, description: String) : this(conference,description,"","","","","",ArrayList())

    companion object {

        val dummyObject: LiveConference
            get() {
                val dummyCon = LiveConference("DummyCon", "Conference McConferenceface")
                dummyCon.groups.add(Group.dummyObject)
                dummyCon.slug = "duco"
                dummyCon.author = ""
                dummyCon.description = "A placeholder conference"
                dummyCon.keywords = ""
                return dummyCon
            }
    }
}
